src/io.ts

```diff
@@ -100,7 +127,7 @@ export const transactionT = t.exact(
     sourceAmount: amountT,
     outputs: t.array(outputT),
     networkByte: t.union([byteT, t.undefined]),
-    data: t.union([t.string, t.undefined]),
+    data: dataT,
```

Comment:
I think would be better to use union of undefined and hex here.

```suggestion
    data: t.union([hexT, t.undefined]),
```

where `hexT` is a 
```
export const hexT = new t.Type<Buffer, string, unknown>(
  'hex',
  t.string.is,
  // `t.success` and `t.failure` are helpers used to build `Either` instances
  (input, context) => {
     if (typeof input !== 'string') {
        return t.failure(input, context, 'not a string');
      }

    try {
        const data = Buffer.from(input, 'hex');
        return t.success(data); // return Buffer! not string
      } catch (e) {
        return t.failure(
          input,
          context,
          'data can not be non-hexadecimal string'
        );
      }
  },
  t.identity
);
```

and [here](https://github.com/VeriBlock/nodecore-js/pull/25/files#diff-8cd4968b81985f0efc2053eabc37db6dR119) write data as it is already Buffer)